File: experiments/llm-reasoning-cost/generated_code/clojure/06_struct_methods_codex.clj
(defrecord Point [^int x ^int y]
  Object
  (squared_mag [this]
    (let [^int x (:x this)
          ^int y (:y this)]
      (int (+ (* x x) (* y y))))))

(defn filter_and_sum
  [points ^int max_mag]
  (reduce
   (fn [^int total ^Point p]
     (let [^int mag (.squared_mag p)]
       (if (<= mag max_mag)
         (int (+ total mag))
         total)))
   0
   points))

File: experiments/llm-reasoning-cost/generated_code/java/06_struct_methods_codex.java
import java.util.List;

public class Main {
    public static class Point {
        int x;
        int y;

        public Point(int x, int y) {
            this.x = x;
            this.y = y;
        }

        public int squared_mag() {
            return x * x + y * y;
        }
    }

    public static int filter_and_sum(List<Point> points, int max_mag) {
        int total = 0;
        for (Point p : points) {
            int mag = p.squared_mag();
            if (mag <= max_mag) {
                total += mag;
            }
        }
        return total;
    }
}